frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { Button, ButtonVariant, Skeleton } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import GenericStatus from '@console/shared/src/components/status/GenericStatus';
+import { VMStatus as VMStatusEnum } from '../../constants/vm/vm-status';
+import { unpauseVMI } from '../../k8s/requests/vmi/actions';
+import { VMStatusBundle } from '../../statuses/vm/types';
+import { getVMStatus } from '../../statuses/vm/vm-status';
+import { VMIKind, VMKind } from '../../types';
+import { hasPendingChanges } from '../../utils/pending-changes';
+import { VmStatusResourcesValue } from './use-vm-status-resources';
+import {
+  getVMILikeLink,
+  getStatusSuffixLabelKey,
+  VMStatusPopoverContent,
+  PendingChangesPopoverContent,
+  ImporterPods,
+  VIEW_POD_LOGS,
+  VIEW_VM_EVENTS,
+  getPodLink,
+  getVMStatusIcon,
+  LinkType,
+} from './vm-status';
+
+import './vm-status.scss';
+
+const PendingChanges: React.FC = () => {
+  const { t } = useTranslation();
+  return <div className="kv-vm-row_status-extra-label">{t('kubevirt-plugin~Pending changes')}</div>;
+};
+
+export const LazyVMStatus: React.FC<LazyVMStatusProps> = ({ vm, vmi, vmStatusResources }) => {
+  const { t } = useTranslation();
+  const vmiLike = vm || vmi;
+  const [vmStatus, setVmStatus] = React.useState<VMStatusBundle>();
+  const { pods, migrations, pvcs, dvs, loaded } = vmStatusResources;
+
+  React.useEffect(() => {
+    if (loaded) {
+      const vmStatusBundle = getVMStatus({
+        vm,
+        vmi,
+        pods,
+        migrations,
+        pvcs,
+        dataVolumes: dvs,
+        vmImports: [],
+      });
+      setVmStatus(vmStatusBundle);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loaded]);
```

Comment:
so the vm status is never updated beside the first one when you load resources ? That does not seem correct

Reply:
you are right, I guess there is no going around listening to all resources? @rawagner

Reply:
there isnt. You can optimize it by using `useMemo` and not the state/effect combo.
You can also try comparing previous state result with current one with deep compare